backends/bmv2/common/midend.cpp

```diff
@@ -105,8 +105,12 @@ MidEnd::MidEnd(CompilerOptions& options) {
         new P4::LocalCopyPropagation(&refMap, &typeMap),
         new P4::ConstantFolding(&refMap, &typeMap),
         new P4::MoveDeclarations(),
-        new P4::ValidateTableProperties({ "implementation", "size", "counters",
-                                          "meters", "support_timeout" }),
+        (options.arch == "v1model") ?
```

Comment:
can you refactor this into subclasses as well?
common should not need to know about various subclasses.
You may want to have different mid-ends for the two back-ends, I think that's reasonable.